### PR TITLE
Uri : rewrite is_windows_file_uri, use a variable instead of regex

### PR DESCRIPTION
**Pull request by glepnir**
@tjdevries Please review. Thanks.

### CODE REVIEW

File: runtime/lua/vim/uri.lua [hunk 1]
```diff
@@ -51,23 +51,31 @@ end
 
 
 --@private
-local function is_windows_file_uri(uri)
-  return uri:match('^file:///[a-zA-Z]:') ~= nil
+-- For test cases, uri is hard-coded.
+-- So control the return value through a variable parameter
+-- If we test a windows uri param is 0
+local function is_windows(...)
```
Comment (vigoux) [inline runtime/lua/vim/uri.lua:57]: I am not sure I understand what you mean here

Comment (glepnir) [reply runtime/lua/vim/uri.lua:57]: Ok .Sorry for my english.. In test file `test/functional/lua/uri_spec.lua`, we test windows uri by pass a hadcored uri test case ` exec_lua([[filepath = 'C:\\Foo\\Bar\\Baz.txt']])` , if we use a variable to judgment current os type will caused the test failed. So when we test windows uri we can pass a value into this function for test.

Comment (vigoux) [reply runtime/lua/vim/uri.lua:57]: Lets have @tjdevries opinion on this, maybe we want to only do those tests on actual windows platforms, and not on linux ?

Comment (glepnir) [reply runtime/lua/vim/uri.lua:57]: Then we don't need to pass in any parameters. Everything will also become cheaper.

**Comment by tjdevries** [comment]
Was there a bug on windows or a performance issue that was causing you to want to do this? I find it a bit confusing and the code harder to follow this way.

**Comment by glepnir** [comment]
@tjdevries  No bugs. because we use a regex function to check windows uri. I think use a variable is better than regex..

**Comment by tjdevries** [comment]
But this won't work if we receive a windows style file path on a unix server (which could be possible I think).

The function should not be passed whether it's a windows path or not. It should determine whether the path is windows for itself.

**Pull request closed without merging**